artemis-protocols/artemis-openwire-protocol/src/main/java/org/apache/activemq/artemis/core/protocol/openwire/OpenWireProtocolManager.java

```diff
@@ -97,6 +100,8 @@
 
    private boolean prefixPacketSize = true;
 
+   private int maxActorSize = 20;
```

Comment:
the actor is in units of commands, I wonder if it would be possible to limit by bytes size and match to the configured serverSocketBuffer size. I guess that raw bytes are already unmarshalled when we are dealing with commands.
It is just that it seems natural to map the pending data on in the broker to what can be pending on the tcp layer. So from a client perspective there is max 2*socketBufferSize bytes pending. Or maybe it is 3x, with one being on the client side of the tcp buffer.

Reply:
I wouldn't know how to do that TBH

Reply:
I was going with 1000, but my computer is slow to sync due to some crypto thing...


I have not decided on the default yet.. but this is mostly about syncs...


the other thing is I will probably make those writes to the journal to not sync if the client is not waiting for a response.